ResearchKit/Common/ORKFormStepViewController.m

```diff
@@ -166,18 +166,6 @@ - (void)addFormItem:(ORKFormItem *)item {
             [(NSMutableArray *)self.items addObject:cellItem];
         }];
         
-    } else if ([[item impliedAnswerFormat] isKindOfClass:[ORKBooleanAnswerFormat class]]) {
-        _hasChoiceRows = YES;
-        {
-            ORKTableCellItem *cellItem = [[ORKTableCellItem alloc] initWithFormItem:item choiceIndex:0];
-            [(NSMutableArray *)self.items addObject:cellItem];
-        }
-        
-        {
-            ORKTableCellItem *cellItem = [[ORKTableCellItem alloc] initWithFormItem:item choiceIndex:1];
-            [(NSMutableArray *)self.items addObject:cellItem];
-        }
-        
```

Comment:
Yep, it's because boolean answer format uses text choice answer format behind the scenes. Since we check for text choice first, the else clause was never reached